loadgen/scenario.go

```diff
@@ -21,8 +23,14 @@ type Scenario struct {
 
 // Executor for a scenario.
 type Executor interface {
+	// PreScenario is called once before the scenario begins execution. Use it for one-time setup,
+	// like adding a custom search attribute.
+	PreScenario(context.Context, ScenarioInfo) error
```

Comment:
They could be put inside of only `GenericExecutor`. Problem being that it's definition of `Run` invokes the user defined run once per iteration, and these need to be called only once each. Doesn't seem like a tremendous complication to me and is generally something that comes up quite a bit, but I can get rid of it from `Executor`.

Reply:
To me:

```go
Executor: ExecutorFunc(func(...) {
  doPreThing()
  defer doPostThing()
  &GenericExecutor{...}.Run()
})
```

makes sense and doesn't infect people that don't need pre/post. At the least I think we should get rid of the generic. I will say I may be less likely to use the generic executor if it looks less generic, but now I'm starting to wonder if we need a generic executor instead of just utilities to run for iterations-or-duration.

Reply:
bro, generics are _more generic_ :wink: 

I would be fine with utilities instead of one generic executor. I'm not going to bother at the moment, because I want to finish this. I'll use a new one for this purpose. I want the storage to live on an actual executor instance so it's not just floating off in global variable space and that seems useful.

At the moment, this is all a question of style. I'd say the generic parameter doesn't really have much of any material impact, it's just a question of taste. Fine, no big deal.

Ideally, these different features you might want in executors could be easily composed. I don't know exactly what that looks like. We can figure it out later.